File: app/src/main/java/de/cineaste/android/viewholder/series/SeasonViewHolder.java
package de.cineaste.android.viewholder.series;

import android.content.Context;
import android.content.res.Resources;
import android.support.v7.widget.RecyclerView;
import android.view.View;
import android.widget.ImageView;
import android.widget.TextView;

import com.squareup.picasso.Picasso;

import java.text.SimpleDateFormat;
import java.util.Date;

import de.cineaste.android.R;
import de.cineaste.android.entity.series.Season;
import de.cineaste.android.listener.ItemClickListener;
import de.cineaste.android.util.Constants;


public class SeasonViewHolder extends RecyclerView.ViewHolder {

    private ImageView poster;
    private TextView seasonNumber;
    private TextView numberOfEpisodes;
    private TextView releaseDate;
    private Context context;
    private Resources resources;
    private ItemClickListener itemClickListener;
    private View view;

    public SeasonViewHolder(View itemView, ItemClickListener listener, Context context) {
        super(itemView);
        poster = itemView.findViewById(R.id.poster_image_view);
        seasonNumber = itemView.findViewById(R.id.season);
        numberOfEpisodes = itemView.findViewById(R.id.episodes);
        releaseDate = itemView.findViewById(R.id.releaseDate);
        this.resources = context.getResources();
        view = itemView;
        itemClickListener = listener;
        this.context = context;
    }

    public void assignData(final Season season) {
        seasonNumber.setText(resources.getString(R.string.currentSeason, String.valueOf(season.getSeasonNumber())));
        numberOfEpisodes.setText(resources.getString(R.string.episodes, String.valueOf(season.getEpisodeCount())));
        if (season.getReleaseDate() == null) {
            releaseDate.setVisibility(View.GONE);
        } else {
            releaseDate.setText(convertDate(season.getReleaseDate()));
        }

        setMoviePoster(season);

        view.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View view) {
                if (itemClickListener != null) {
                    itemClickListener.onItemClickListener(season.getId(), new View[] {view, poster});
                }
            }
        });

    }

    private void setMoviePoster(Season season) {
        String posterName = season.getPosterPath();
        String posterUri =
                Constants.Companion.getPOSTER_URI_SMALL()
                        .replace("<posterName>", posterName != null ? posterName : "/")
                        .replace("<API_KEY>", context.getString(R.string.movieKey));
        Picasso.with(context).load(posterUri).resize(342, 513).error(R.drawable.placeholder_poster).into(poster);
    }

    private String convertDate(Date date) {
        SimpleDateFormat simpleDateFormat = new SimpleDateFormat("dd.MM.yyyy", resources.getConfiguration().locale);
        return simpleDateFormat.format(date);
    }
}

File: app/src/main/kotlin/de/cineaste/android/viewholder/series/SeasonViewHolder.kt
package de.cineaste.android.viewholder.series

import android.content.Context
import android.content.res.Resources
import androidx.recyclerview.widget.RecyclerView
import android.view.View
import android.widget.ImageView
import android.widget.TextView

import com.squareup.picasso.Picasso

import java.text.SimpleDateFormat
import java.util.Date
import java.util.Locale

import de.cineaste.android.R
import de.cineaste.android.entity.series.Season
import de.cineaste.android.listener.ItemClickListener
import de.cineaste.android.util.Constants

class SeasonViewHolder(
    private val view: View,
    private val itemClickListener: ItemClickListener?,
    private val context: Context
) : RecyclerView.ViewHolder(view) {

    private val poster: ImageView = view.findViewById(R.id.poster_image_view)
    private val seasonNumber: TextView = view.findViewById(R.id.season)
    private val numberOfEpisodes: TextView = view.findViewById(R.id.episodes)
    private val releaseDate: TextView = view.findViewById(R.id.releaseDate)
    private val resources: Resources = context.resources

    fun assignData(season: Season) {
        seasonNumber.text =
            resources.getString(R.string.currentSeason, season.seasonNumber.toString())
        numberOfEpisodes.text =
            resources.getString(R.string.episodes, season.episodeCount.toString())
        if (season.releaseDate == null) {
            releaseDate.visibility = View.GONE
        } else {
            releaseDate.text = convertDate(season.releaseDate)
        }

        setMoviePoster(season)

        view.setOnClickListener { view ->
            itemClickListener?.onItemClickListener(season.id, arrayOf(view, poster))
        }
    }

    private fun setMoviePoster(season: Season) {
        val posterName = season.posterPath
        val posterUri = Constants.POSTER_URI_SMALL
            .replace("<posterName>", posterName ?: "/")
            .replace("<API_KEY>", context.getString(R.string.movieKey))
        Picasso.get()
            .load(posterUri)
            .resize(342, 513)
            .error(R.drawable.placeholder_poster)
            .into(poster)
    }

    private fun convertDate(date: Date?): String {
        val simpleDateFormat = SimpleDateFormat("dd.MM.yyyy", Locale.ENGLISH)
        return simpleDateFormat.format(date)
    }
}
